tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,420 @@
+                                      enable_authentication=self.enable_authentication,
+                                      conn_disconn_timeout=self.connection_disconnect_timeout)
+
+        # Validate Mqtt __init__ called with following parameters
+        mock_init.assert_called_with(self.url, self.port, self.identity, self.tls_conf,
+                                     self.qos_details, self.client_id, self.clean_session, mqtt_dcc_comms.userdata,
+                                     self.protocol, self.transport, self.keep_alive, self.enable_authentication,
+                                     self.connection_disconnect_timeout)
+
+    @mock.patch.object(Mqtt, 'disconnect')
+    def test_disconnect(self, mock_disconnect):
+        """
+        Test MqttDccComms _disconnect method
+        :param mock_disconnect: Mocked MQTT disconnect method
+        :return: None
+        """
+        # Call MqttDccComms _disconnect
+        self.client._disconnect()
+
+        # Check call made to the Mqtt disconnect method
+        mock_disconnect.assert_called()
+
+    @mock.patch.object(Mqtt, 'publish')
+    def test_send_with_msg_attr(self, mock_publish):
+        """
+        Test MqttDccComms publish method with messaging attributes
+        :param mock_publish: Mocked MQTT publish method
+        :return: None
+        """
+
+        # Create subscribe MqttMessagingAttributes
+        mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic", sub_topic="subscribe-topic", sub_qos=1,
+                                                sub_callback=callback_function, pub_retain=False)
+
+        # Call MqttDccComms send method
+        self.client.send(message=self.publish_message, msg_attr=self.mqtt_msg_attr)
+
+        # Check implementation calling Mqtt publish method with following params
+        mock_publish.assert_called_with(mqtt_msg_attr.pub_topic, self.publish_message,
+                                        mqtt_msg_attr.pub_qos, mqtt_msg_attr.pub_retain)
+
+    @mock.patch.object(Mqtt, 'publish')
+    def test_send_without_msg_attr(self, mock_publish):
+        """
+        Test MqttDccComms publish method without messaging attributes
+        :param mock_publish: Mocked MQTT publish method
+        :return: None
+        """
+        # Call MqttDccComms send method
+        self.client.send(message=self.publish_message, msg_attr=None)
+
+        # Check implementation calling Mqtt publish method with following params
+        mock_publish.assert_called_with(self.client.msg_attr.pub_topic, self.publish_message,
+                                        self.client.msg_attr.pub_qos, self.client.msg_attr.pub_retain)
+
+    @mock.patch.object(Mqtt, 'subscribe')
+    def test_receive(self, mocked_subscribe):
+        """
+        Test MqttDccComms receive method implementation with msg_attr.
+        :return: None 
+        """
+
+        # Call receive with mqtt_msg_attr
+        self.client.receive(self.mqtt_msg_attr)
```

Comment:
Create a different mqtt_msg_attr with different subscribe related attributes than self.mqtt_msg_attr and pass it to client.receive().

We are testing if receive is using the passed mqtt_msg_attr.  If we pass self.mqtt_msg_attr here, test case is compromised.